File: sources/LiquidSense Release/me/aquavit/liquidsense/command/commands/SetNameCommand.java
package me.aquavit.liquidsense.command.commands;

import me.aquavit.liquidsense.LiquidSense;
import me.aquavit.liquidsense.command.Command;
import me.aquavit.liquidsense.module.Module;
import me.aquavit.liquidsense.module.ModuleInfo;
import net.minecraft.util.EnumChatFormatting;

import java.util.ArrayList;
import java.util.List;
import java.util.stream.Collectors;

public class SetNameCommand extends Command {
    public SetNameCommand() {
        super("name");
    }

    @Override
    public void execute(final String[] args) {
        if (args.length > 1) {
            String arg = args[1].toLowerCase();

            switch (arg) {
                case "load":
                    LiquidSense.fileManager.loadConfig(LiquidSense.fileManager.setNameConfig);
                    break;
                case "list":
                    for (Module module : LiquidSense.moduleManager.getModules()) {
                        if (module.getArrayListName().equals(module.getClass().getAnnotation(ModuleInfo.class).name())) continue;
                        chat("Module <" + module.getArrayListName() + "> is §7" + module.getClass().getAnnotation(ModuleInfo.class).name());
                    }
                    break;
                case "cleaner":
                    for (Module module : LiquidSense.moduleManager.getModules()) {
                        if (args.length > 2) {
                            String newValue = getName(args, 2);

                            Module oldModule = LiquidSense.moduleManager.getModule(newValue);
                            if (oldModule == null) return;

                            if (module.getName().equalsIgnoreCase(oldModule.getName())) {
                                module.setArrayListName(module.getClass().getAnnotation(ModuleInfo.class).name());

                                LiquidSense.fileManager.saveConfig(LiquidSense.fileManager.setNameConfig);
                                chat("reset Module <" + newValue + '>');
                                return;
                            }
                        }
                        module.setArrayListName(module.getClass().getAnnotation(ModuleInfo.class).name());
                        LiquidSense.fileManager.saveConfig(LiquidSense.fileManager.setNameConfig);
                    }
                    chat("reset Module name ");
                    break;
                default:
                    if (args.length > 2) {
                        String newValue = getName(args, 2);
                        String oldVaule = getName(args, 1);
                        Module module = LiquidSense.moduleManager.getModule(oldVaule);
                        if (module == null) {
                            chat("Module §a§l" + args[1] + "§3 not found.");
                            return;
                        }
                        module.setNameCommand(newValue);
                        LiquidSense.fileManager.saveConfig(LiquidSense.fileManager.setNameConfig);
                        chat("Module §a§l" + args[1] + "§l§a to " + newValue);
                        return;
                    }
                    break;
            }
            return;
        }
        chatSyntax(new String[]{"<module> <name>", "<cleaner/load/list>"});
    }

    @Override
    public List<String> tabComplete(String[] args) {
        if (args.length == 0) return new ArrayList<>();

        String moduleName = args[0];

        if (args.length == 1) {
            return LiquidSense.moduleManager.getModules().stream()
                    .map(Module::getName)
                    .filter(module -> module.toLowerCase().startsWith(moduleName.toLowerCase()))
                    .collect(Collectors.toList());
        }
        return new ArrayList<>();
    }

    public String getName(final String[] args, int size) {
        String string = args[size];
        switch (string) {
            case "#":
                string = string.replace("#", "");
                break;
            case "#l":
                string = string.replace("#l", EnumChatFormatting.BOLD.toString());
                break;
            case "#o":
                string = string.replace("#o", EnumChatFormatting.ITALIC.toString());
                break;
            case "#m":
                string = string.replace("#m", EnumChatFormatting.STRIKETHROUGH.toString());
                break;
            case "#n":
                string = string.replace("#n", EnumChatFormatting.UNDERLINE.toString());
                break;
        }
        return string;
    }
}

File: sources/LiquidSense Dev/net/ccbluex/liquidbounce/features/command/commands/SetNameCommand.kt
package net.ccbluex.liquidbounce.features.command.commands

import com.google.common.base.Splitter
import com.google.gson.Gson
import com.google.gson.JsonArray
import com.google.gson.JsonObject
import net.ccbluex.liquidbounce.LiquidBounce
import net.ccbluex.liquidbounce.event.EventManager
import net.ccbluex.liquidbounce.features.command.Command
import net.ccbluex.liquidbounce.features.module.ModuleInfo
import net.ccbluex.liquidbounce.file.FileManager
import net.minecraft.util.EnumChatFormatting
import org.apache.commons.io.FileUtils
import org.apache.commons.lang3.StringUtils

class SetNameCommand : Command("name") {


    override fun execute(args: Array<String>) {

        if (args.size > 1) {
            when {
                args[1].equals("load", true) -> {
                    LiquidBounce.fileManager.loadConfig(LiquidBounce.fileManager.setnameConfig)
                    return
                }
                args[1].equals("list", true) -> {
                    for (module in LiquidBounce.moduleManager.modules) {
                        if (module.arrayListName.equals(module::class.java.getAnnotation(ModuleInfo::class.java).name))
                            continue

                        chat("Module <" + module.arrayListName + "> is §7" + module::class.java.getAnnotation(ModuleInfo::class.java).name)
                    }
                    return
                }
                args[1].equals("cleaner", true) -> {
                    for (module in LiquidBounce.moduleManager.modules) {
                        if (args.size > 2) {

                            var newValue = getname(args, 2)

                            val oldmodule = LiquidBounce.moduleManager.getModule(newValue) ?: return


                            if (module.name.equals(oldmodule.name)) {
                                module.arrayListName = module::class.java.getAnnotation(ModuleInfo::class.java).name

                                LiquidBounce.fileManager.saveConfig(LiquidBounce.fileManager.setnameConfig)
                                chat("reset Module <$newValue>")
                                return
                            }
                        } else {
                            module.arrayListName = module::class.java.getAnnotation(ModuleInfo::class.java).name
                            LiquidBounce.fileManager.saveConfig(LiquidBounce.fileManager.setnameConfig)
                        }
                    }
                    chat("reset Module name ")
                    return
                }

                else -> {
                    if (args.size > 2) {

                        var newValue = getname(args, 2)
                        var oldVaule = getname(args, 1)

                        // Get module by name
                        val module = LiquidBounce.moduleManager.getModule(oldVaule)

                        if (module == null) {
                            chat("Module §a§l" + args[1] + "§3 not found.")
                            return
                        }

                        module.setNameCommad(newValue)
                        LiquidBounce.fileManager.saveConfig(LiquidBounce.fileManager.setnameConfig)
                        chat("Module §a§l" + args[1] + "§l§a to " + newValue)
                        return
                    }
                }
            }
        }
        chatSyntax(arrayOf("<module> <name>", "<cleaner/load/list> "))
    }

    override fun tabComplete(args: Array<String>): List<String> {
        if (args.isEmpty()) return emptyList()

        val moduleName = args[0]

        return when (args.size) {
            1 -> LiquidBounce.moduleManager.modules
                .map { it.name }
                .filter { it.startsWith(moduleName, true) }
                .toList()
            else -> emptyList()
        }
    }

    fun getname(args: Array<String>, size: Int): String {
       return when {
            args[size].contains("#") -> args[size].replace("#", " ")
            args[size].contains("#l") -> args[size].replace("#l", EnumChatFormatting.BOLD.toString())
            args[size].contains("#o") -> args[size].replace("#o", EnumChatFormatting.ITALIC.toString())
            args[size].contains("#m") -> args[size].replace("#m", EnumChatFormatting.STRIKETHROUGH.toString())
            args[size].contains("#n") -> args[size].replace("#n", EnumChatFormatting.UNDERLINE.toString())
            else -> args[size]
        }
    }
}
